packages/quantic/force-app/main/default/lwc/quanticPager/e2e/quantic-pager.e2e.ts

```diff
@@ -0,0 +1,229 @@
+        await expect(pager.previousPageButton).not.toBeVisible();
+        await expect(pager.nextPageButton).not.toBeVisible();
+        await expect(pager.pageButtons).not.toBeVisible();
+
+        resumeSearchRequest();
+
+        await expect(pager.previousPageButton).toBeVisible();
+        await expect(pager.nextPageButton).toBeVisible();
+        await expect(pager.pageButtons).not.toBeNull();
+      });
+    });
+
+    test.describe('when on the first page', () => {
+      test('should disable the previous page button', async ({pager}) => {
+        await expect(pager.pageButtons.nth(0)).toHaveAttribute(
+          'aria-pressed',
+          'true'
+        );
+        await expect(pager.previousPageButton).toBeDisabled();
+      });
+    });
+
+    test.describe('when on the last page', () => {
+      test.beforeEach(async ({pager, search}) => {
+        await pager.goToLastPage();
+        await search.waitForSearchResponse();
+      });
+
+      test('should disable the next page button', async ({pager}) => {
+        await expect(pager.pageButtons.nth(-1)).toHaveAttribute(
+          'aria-pressed',
+          'true'
+        );
+        await expect(pager.nextPageButton).toBeDisabled();
+      });
+    });
+
+    test.describe('when clicking the next page button', () => {
+      test('should trigger a new search and log analytics', async ({
+        pager,
+        search,
+      }) => {
+        await pager.clickNextPageButton();
+        const response = await search.waitForSearchResponse();
+        const {firstResult} = await response.request().postDataJSON();
+        await expect(firstResult).toBe(numberOfResultsPerPage);
+        await pager.waitForPagerUaAnalytics('pagerNext');
+      });
+    });
+
+    test.describe('when clicking the previous page button', () => {
+      test('should trigger a new search and log analytics', async ({
+        pager,
+        search,
+      }) => {
+        await pager.clickNextPageButton();
+        await search.waitForSearchResponse();
+
+        await pager.clickPreviousPageButton();
+        const response = await search.waitForSearchResponse();
+        const {firstResult} = await response.request().postDataJSON();
+        await expect(firstResult).toBe(0);
+        await pager.waitForPagerUaAnalytics('pagerPrevious');
+      });
+    });
+
+    test.describe('when clicking a specific page button', () => {
+      test('should trigger a new search and log analytics', async ({
+        pager,
+        search,
+      }) => {
+        const examplePage = 3;
+        await pager.clickPageNumberButton(examplePage);
+        const response = await search.waitForSearchResponse();
+        const {firstResult} = await response.request().postDataJSON();
+        await expect(firstResult).toBe(
+          numberOfResultsPerPage * (examplePage - 1)
+        );
+        await pager.waitForPagerUaAnalytics('pagerNumber');
+      });
+    });
+
+    test.describe('when a new search is made', () => {
```

Comment:
Not sure that this should be here either, I think this is more of a full search page kinda test, than specific to the pager.

Reply:
After live discussion, these kind of interactions should be kept at a component level.